sycl/source/detail/builtins_relational.cpp

```diff
@@ -458,9 +458,15 @@ __SYCL_EXPORT rel_res_t sycl_host_SignBitSet(s::cl_float x) __NOEXC {
 __SYCL_EXPORT rel_res_t sycl_host_SignBitSet(s::cl_double x) __NOEXC {
   return std::signbit(x);
 }
+#ifndef __GNUC__
+// GCC 11.3.0 (and friends) has an internal compiler error when this function is
+// passed to the macro declaration below. ( MAKE_1V ... 4 ... ) Fortunately, we
+// don't actually need it when compiling the SYCL library itself.
+// TODO: switch to templates instead of these overload expansions.
 __SYCL_EXPORT s::cl_int __vSignBitSet(s::cl_float x) __NOEXC {
   return -static_cast<s::cl_int>(std::signbit(x));
 }
+#endif
```

Comment:
Discussed offline. This is in library source code and is likely used but is missing test coverage, so we would be breaking ABI by conditionally excluding it here.

Reply:
@steffenlarsen  - the ABI symbol table is updated to reflect that, and it's not the only ABI change in this PR. Users will definitely have to recompile. Attempting to get a dispensation.

I checked this out yesterday, and this exact case ( vec<float> -> int)  IS under test. I think it passes because there are other overloads and the compiler is implicitly converting. The overloads are all identical.  Ultimately, I think this OK. Let me know if you disagree - I can make sycl::signbitset a template and move it into the headers. Just need to find a place to sequester a solitary math built-in.  I think the better solution is to, one day, move all the math built-ins to templates and lose these macro expansions. But that's out of scope for this.

Reply:
You might be right. It could be that we have an implicit promotion of `cl_float` to `cl_double` and then a demotion from `cl_long` to `cl_int`. I might worry that this could cause warnings on some compilers, so you might need to be ready for promotion/demotion warnings in post-commit.

Looking at it, I don't see why we would even export the `__v*` functions. Seems to me like they are simply used inside other exported functions. Had the ABI-break window not been closed we should've removed `__SYCL_EXPORT` from these. Next time, maybe!

> the ABI symbol table is updated to reflect that, and it's not the only ABI change in this PR. Users will definitely have to recompile. Attempting to get a dispensation.

Assuming you are right and we can safely mask this out, wouldn't the symbols check when we compile the library with clang (e.g. in post-commit) as the symbol would suddenly be back?